app/models/manageiq/providers/ansible_tower/shared/automation_manager/credential.rb

```diff
@@ -8,32 +8,62 @@ def create_in_provider(manager_id, params)
         connection.api.credentials.create!(params)
       end
 
+      refresh(manager)
+      find_by!(:resource_id => manager.id, :manager_ref => credential.id)
+    end
+
+    def create_in_provider_queue(manager_id, params)
+      manager = ExtManagementSystem.find(manager_id)
+      queue(manager.my_zone, "create_in_provider", [manager_id, params], "Creating #{name}")
+    end
+
+    private
+
+    def refresh(manager)
       # Get the record in our database
       # TODO: This needs to be targeted refresh so it doesn't take too long
       task_ids = EmsRefresh.queue_refresh_task(manager)
       task_ids.each { |tid| MiqTask.wait_for_taskid(tid) }
-
-      find_by!(:resource_id => manager.id, :manager_ref => credential.id)
     end
 
-    def create_in_provider_queue(manager_id, params)
+    def queue(my_zone, method_name, args, action)
       task_opts = {
-        :action => "Creating #{name}",
+        :action => action,
         :userid => "system"
       }
 
-      manager = ExtManagementSystem.find(manager_id)
-
       queue_opts = {
-        :args        => [manager_id, params],
+        :args        => args,
         :class_name  => name,
-        :method_name => "create_in_provider",
+        :method_name => method_name,
```

Comment:
This won't work for the instance methods.  You'll have to pass in the `id` of the current instance in order to have it look up the instance to call the method.